enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -89,27 +89,77 @@ to request that a set of container images are pinned and pre-loaded:
     metadata:
       name: my-pinned-images
     spec:
-      nodeSelector:
-        matchLabels:
-          node-role.kubernetes.io/control-plane: ""
       pinnedImages:
-      - quay.io/openshift-release-dev/ocp-release@sha256:...
-      - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
-      - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+      - name: "quay.io/openshift-release-dev/ocp-release@sha256:..."
+      - name: "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:..."
+      - name: "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:..."
       ...
     ```
 
-1. A new `PinnedImageSetController` sub-controller of the
-_machine-config-controller_ ensures that all the images are pinned and
-pulled in all the nodes that match the node selector.
+1. `MachineConfigPoolSpec` will add an optional field taking a reference to a `PinnedImageSet`.
+  
+  ```yaml
+  apiVersion: machineconfiguration.openshift.io/v1
+  kind: MachineConfigPool
+  spec:
+    pinnedImageSets:
+    - name: "my-pinned-images"
+  ```
+
+1. A new `Validation WebHook` will be added to ensure the validity of
```

Comment:
for these MCP changes, will they be going into the techpreview MCP CRD as well? That would probably make sense.

Reply:
correct